app/src/test/kotlin/fr/free/nrw/commons/contributions/ContributionsPresenterTest.kt

```diff
@@ -27,84 +37,58 @@ class ContributionsPresenterTest {
 
     lateinit var loader: Loader<Cursor>
 
+    lateinit var liveData: LiveData<List<Contribution>>
+
+    @Rule @JvmField var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    lateinit var scheduler : Scheduler
+
     /**
      * initial setup
      */
     @Before
     @Throws(Exception::class)
     fun setUp() {
         MockitoAnnotations.initMocks(this)
+        scheduler=TestScheduler()
         cursor = Mockito.mock(Cursor::class.java)
         contribution = Mockito.mock(Contribution::class.java)
-        contributionsPresenter = ContributionsPresenter(repository)
+        contributionsPresenter = ContributionsPresenter(repository,scheduler,scheduler)
         loader = Mockito.mock(CursorLoader::class.java)
         contributionsPresenter?.onAttachView(view)
+        liveData=MutableLiveData()
     }
 
-
     /**
-     * Test presenter actions onGetContributionFromCursor
+     * Test fetch contributions
      */
     @Test
-    fun testGetContributionFromCursor() {
-        contributionsPresenter?.getContributionsFromCursor(cursor)
-        verify(repository)?.getContributionFromCursor(cursor)
+    fun testFetchContributions(){
+        Mockito.`when`(repository?.getString(ArgumentMatchers.anyString())).thenReturn("10")
+        Mockito.`when`(repository?.fetchContributions()).thenReturn(liveData)
+        contributionsPresenter?.fetchContributions()
+        verify(repository)?.fetchContributions()
     }
```

Comment:
This test barely covers this method and the coverage overall of the class is far from full , no `view` interactions, deleteUpload or getItemAtPosition

Reply:
True, this test does not justify the coverage required, someone volunteer or I will raise, else will raise a separate PR to cover the test case. In the current one, I had actually aimed not to miss the existing test cases.

Reply:
Cool